xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp

```diff
@@ -100,23 +100,41 @@ bool CGUIDialogPVRGuideInfo::ActionCancelTimer(CFileItemPtr timer)
     return bReturn;
   }
 
-  // prompt user for confirmation of timer deletion
-  CGUIDialogYesNo* pDialog = (CGUIDialogYesNo*)g_windowManager.GetWindow(WINDOW_DIALOG_YES_NO);
-
-  if (pDialog)
+  if(!timer->GetPVRTimerInfoTag()->m_bIsRepeating)
   {
-    pDialog->SetHeading(265);
-    pDialog->SetLine(0, "");
-    pDialog->SetLine(1, timer->GetPVRTimerInfoTag()->m_strTitle);
-    pDialog->SetLine(2, "");
-    pDialog->DoModal();
+    // prompt user for confirmation of timer deletion
+    CGUIDialogYesNo* pDialog = (CGUIDialogYesNo*)g_windowManager.GetWindow(WINDOW_DIALOG_YES_NO);
 
-    if (pDialog->IsConfirmed())
+    if (pDialog)
     {
-      Close();
-      bReturn = CPVRTimers::DeleteTimer(*timer);
+      pDialog->SetHeading(265);
+      pDialog->SetLine(0, "");
+      pDialog->SetLine(1, timer->GetPVRTimerInfoTag()->m_strTitle);
+      pDialog->SetLine(2, "");
+      pDialog->DoModal();
+
+      if (pDialog->IsConfirmed())
+      {
+        Close();
+        bReturn = CPVRTimers::DeleteTimer(*timer);
+      }
     }
   }
+  else
+  {
+    bool bCancel(false);
+    bool bConfirm(false);
+
+    /* prompt user for deleting the complete schedule */
+    bConfirm = CGUIDialogYesNo::ShowAndGetInput(g_localizeStrings.Get(122),g_localizeStrings.Get(803),"",
+        StringUtils::Format(g_localizeStrings.Get(824).c_str(), timer->GetPVRTimerInfoTag()->GetType().c_str()), bCancel);
+
+    if (!bCancel)
+      Close();
+
+    if(bConfirm && !bCancel)
+      bReturn = CPVRTimers::DeleteTimer(*timer, false, true);
```

Comment:
Since this is the only place the third parameter (`bDeleteSchedule`) is set to true I think it should be factored out to a separate method, i.e. `CPVRTimers::DeleteSchedule(const CPVRTimerInfoTagPtr &timer)` or something. This way you don't have to add the `bDeleteSchedule` bool to all related methods.